__test__/api/replies/deleteSinglereply.test.js

```diff
@@ -0,0 +1,120 @@
+const app = require("../../../server");
+const supertest = require("supertest");
+const ReplyModel = require("../../../models/replies");
+const CommentModel = require("../../../models/comments");
+const commentHandler = require("../../../utils/commentHandler");
+const replyHandler = require("../../../utils/replyHandler");
+const request = supertest(app);
+
+describe("DELETE /comments/:commentId/replies/:replyId", () => {
+  let comment;
+  let reply;
+
+  beforeEach(async () => {
+    comment = new CommentModel({
+      content: "A comment from user 1",
+      ownerId: "user1@email.com",
+      origin: "123123",
+      refId: 2,
+      applicationId: global.application._id,
+    });
+
+     reply= new ReplyModel({
+      content: "A reply from user 2",
+      ownerId: "user2@email.com",
+      commentId: comment.id,
+    });
+
+    //save dummy comment
+    await comment.save();
+    await reply.save();
+
+    // Add replies to the dummy comment
+    await CommentModel.findByIdAndUpdate(comment.id, {replies: reply.id});
+
+    // Cache response objects
+    // comment = commentHandler(dcomment);
+    // reply = replyHandler(dreply1);
+  });
+
+  afterEach(async () => {
+    // Delete mocks from the database.
+    await CommentModel.findByIdAndDelete(comment.id);
+    await ReplyModel.findByIdAndDelete(reply.id);
+
+    // Delete cache.
+    comment = null;
+    reply = null;
+  });
+
+  //200 success
+  test("should delete a single reply", async () => {
+    const expectedC = commentHandler(comment);
+    const expectedR = replyHandler(reply);
+    const url = `/v1/comments/${expectedC.commentId}/replies/${expectedR.replyId}`;
+    const bearerToken = `bearer ${global.appToken}`;
+
+    const res = await request
+      .delete(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: expectedR.ownerId,
+      });
+    expect(res.status).toBe(200);
+    expect(res.body.status).toEqual("success");
+    expect(res.body.data).toEqual({
```

Comment:
add `matchers` to check db that comment no longer has deleted replies, something like..

```
res = DELETE /reply/url
comment = await model.findbyid(commentId)
expect(comment.replies).notContains(replyId)
```